File: 0035-search-insert-position/solution.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        
        int min=0,max=nums.size()-1,ans;
        while(min<=max){
            ans = (max+min)/2;
            if(nums[ans] == target)return ans;
            else if(nums[ans] > target) max = ans-1;
            else min = ans+1;
        }
        
        return min;
    }
};

File: 0035-search-insert-position/solution.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        
        MIN = 0
        MAX = len(nums)
        
        if nums[-1] < target:
          return len(nums)
        
        while MIN <= MAX:
          i = int((MIN+MAX)/2)
        
          if nums[i] == target:
            return i  
          
          if nums[i] > target:
            MAX = i - 1

          if nums[i] < target:
            MIN = i + 1
          
        return MIN
